airflow/contrib/hooks/gcp_vision_hook.py

```diff
@@ -0,0 +1,253 @@
+                self.log.error('The request failed:\n%s', str(e))
+                raise AirflowException(e)
+        except RetryError as e:
+            self.log.error('The request failed due to a retryable error and retry '
+                           'attempts failed.')
+            raise AirflowException(e)
+        except ValueError as e:
+            self.log.error('The request failed, the parameters are invalid.')
+            raise AirflowException(e)
+
+    @staticmethod
+    def _get_entity_name(is_product, project_id, location, entity_id):
+        if is_product:
+            return ProductSearchClient.product_path(project_id, location, entity_id)
+        else:
+            return ProductSearchClient.product_set_path(project_id, location, entity_id)
+
+    @staticmethod
+    def _determine_name(entity, entity_id, location, project_id):
+        is_product = type(entity) == Product
+        entity_label = 'Product' if is_product else 'ProductSet'
+        entity_id_label = 'product_id' if is_product else 'productset_id'
+        explicit_name = getattr(entity, 'name')
+        if not location or not entity_id:
+            # Not enough parameters to construct the name. Trying to use the name from Product / ProductSet.
+            if not explicit_name:
+                raise AirflowException("Unable to determine the {} name. Please either set the name "
+                                       "directly in the {} object or provide the `location` and "
+                                       "`{}` parameters.".format(entity_label, entity_label, entity_id_label))
+        else:
+            template_name = CloudVisionHook._get_entity_name(is_product, project_id, location, entity_id)
+            if not explicit_name:
+                entity.name = template_name
+            elif explicit_name != template_name:
+                raise AirflowException("The {} name provided in the object ({}) is different than the"
+                                       " name created from the input parameters ({}). Please either: "
+                                       "1) Remove the {} name, 2) Remove the location and "
+                                       "{} parameters, 3) Unify the {} name and input "
+                                       "parameters.".format(entity_label, explicit_name, template_name,
+                                                            entity_label, entity_id_label, entity_label))
+        return entity
+
+    @staticmethod
+    def _get_autogenerated_id(response):
+        try:
+            name = response.name
+        except AttributeError as e:
+            raise AirflowException('Unable to get name from response... [{}]\n{}'.format(response, e))
+        match = re.search(r"[^/]*$", name)
+        if match and match.group(0):
+            product_set_id = match.group(0)
+        else:
+            raise AirflowException('Unable to get id from name... [{}]'.format(name))
+        return product_set_id
```

Comment:
If I understand it correctly, you can shorten it to
name = response.get('name')
if '/' not in name:
            raise AirflowException('Unable to get name from response... [{}]\n{}'.format(response, e))
return name.rsplit('/', 1)[1]